spec/abci++/abci++_methods.md

```diff
@@ -334,26 +425,23 @@ title: Methods
       proposed block.
     * `RequestPrepareProposal` contains a preliminary set of transactions `txs` that Tendermint
       retrieved from the mempool, called _raw proposal_. The Application can modify this
-      set via `ResponsePrepareProposal.tx_records` (see [TxRecord](#txrecord)).
+      set and return a modified set of transactions via `ResponsePrepareProposal.txs` .
         * The Application _can_ modify the raw proposal: it can reorder, remove or add transactions.
-          Let `tx` be a transaction in `txs`:
+          Let `tx` be a transaction in `txs` (set of transactions within `RequestPrepareProposal`):
             * If the Application considers that `tx` should not be proposed in this block, e.g.,
               there are other transactions with higher priority, then it should not include it in
-              `tx_records`. In this case, Tendermint will not remove `tx` from the mempool. The
+              `ResponsePrepareProposal.txs`. However, Tendermint will not remove `tx` from the mempool. The
               Application should be extra-careful, as abusing this feature may cause transactions
-              to stay much longer than needed in the mempool.
-            * If the Application considers that `tx` should not be included in the proposal and
-              removed from the mempool, then the Application should include it in `tx_records` and
-              _mark_ it as `REMOVED`. In this case, Tendermint will remove `tx` from the mempool.
+              to stay much longer than needed in the mempool.         
```

Comment:
Isn't it still the case that if applications keep removing transactions they keep them in the mempool but they are constantly not executed. THis can also potentially fill up the mempool with transactions that keep getting dropped. Maybe the wording can be different?

Reply:
Yes, that is true, but the reason for including that sentence was to encourage the usage of REMOVE. Now, the App has no other option, so I'm not sure it can be "abused".
I'd be OK with keeping it, but then we need to reword it as you suggest.

Reply:
I will rephrase it a bit but I still think it is useful to make the user aware of the potential drawback of removing transactions.